tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/Tests/ChoosePropertyTests/ChooseProperty.test.dialog

```diff
@@ -0,0 +1,84 @@
+﻿{
+  "$schema": "../../../tests.schema",
+  "$kind": "Microsoft.Test.Script",
+  "description": "Tests for Choosing Property",
+  "httpRequestMocks": [
+    "LuisChooseProperty.mock"
+  ],
+  "dialog": {
+    "$kind": "Microsoft.AdaptiveDialog",
+    "recognizer": {
+      "$kind": "Microsoft.LuisRecognizer",
+      "applicationId": "00000000-0000-0000-0000-000000000000",
+      "endpointKey": "00000000000000000000000000000000",
+      "endpoint": "https://westus.api.cognitive.microsoft.com",
+      "predictionOptions": {
+        "IncludeAPIResults": true
+      }
+    },
+    "schema": "threeProperties.json",
+    "triggers": [
+      {
+        "$kind": "Microsoft.OnBeginDialog",
+        "actions": [
+          {
+            "$kind": "Microsoft.SendActivity",
+            "activity": "welcome"
+          }
+        ]
+      },
+      {
+        "$kind": "Microsoft.OnEndOfActions",
+        "condition": "=!$Name",
+        "priority": 0,
+        "actions": [
+          {
+            "$kind": "Microsoft.Ask",
+            "activity": "Name?",
+            "expectedProperties": [
+              "Name"
+            ]
+          }
+        ]
+      },
+      {
+        "$kind": "Microsoft.OnChooseProperty",
+        "actions": [
+          {
+            "$kind": "Microsoft.Ask",
+            "activity": "Meat or Cheese ?",
+            "expectedProperties": [
+              "PropertyToChange"
+            ]
+          }
+        ]
+      }
+    ]
+  },
+  "script": [
+    {
+      "$kind": "Microsoft.Test.UserConversationUpdate",
+      "membersAdded": [
+        "Bot",
+        "User"
+      ],
+      "membersRemoved": []
+    },
+    {
+      "$kind": "Microsoft.Test.AssertReply",
+      "text": "welcome"
+    },
+    {
+      "$kind": "Microsoft.Test.AssertReply",
+      "text": "Name?"
+    },
+    {
+      "$kind": "Microsoft.Test.UserSays",
+      "text": "none"
+    },
+    {
+      "$kind": "Microsoft.Test.AssertReply",
+      "text": "Meat or Cheese ?"
+    }
```

Comment:
Would be good to extend script to choosing one and making sure it gets set.

Reply:
Add memory assertions

---
In reply to: [491131136](https://github.com/microsoft/botbuilder-dotnet/pull/4640#discussion_r491131136) [](ancestors = 491131136)